Sender-Stats/Sender-Stats/ChatViewController.swift

```diff
@@ -0,0 +1,299 @@
+        return subscribers.count + 1
+    }
+
+    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "videoCell", for: indexPath)
+        let videoView: UIView? = {
+            if (indexPath.row == 0) {
+                return publisher.view
+            } else {
+                let sub = subscribers[indexPath.row - 1]
+                return sub.view
+            }
+        }()
+
+        if let viewToAdd = videoView {
+            viewToAdd.frame = cell.bounds
+            cell.addSubview(viewToAdd)
+        }
+        return cell
+    }
+}
+
+// MARK: - OTSession delegate callbacks
+extension ChatViewController: OTSessionDelegate {
+    func sessionDidConnect(_ session: OTSession) {
+        print("Session connected")
+        doPublish()
+    }
+
+    func sessionDidDisconnect(_ session: OTSession) {
+        print("Session disconnected")
+    }
+
+    func session(_ session: OTSession, streamCreated stream: OTStream) {
+        print("Session streamCreated: \(stream.streamId)")
+        doSubscribe(stream)
+    }
+
+    func session(_ session: OTSession, streamDestroyed stream: OTStream) {
+        print("Session streamDestroyed: \(stream.streamId)")
+        cleanupSubscriber(stream)
+    }
+
+    func session(_ session: OTSession, didFailWithError error: OTError) {
+        print("session Failed to connect: \(error.localizedDescription)")
+    }
+}
+
+// MARK: - OTPublisher delegate callbacks
+extension ChatViewController: OTPublisherDelegate {
+    func publisher(_ publisher: OTPublisherKit, streamCreated stream: OTStream) {
+    }
+
+    func publisher(_ publisher: OTPublisherKit, streamDestroyed stream: OTStream) {
+    }
+
+    func publisher(_ publisher: OTPublisherKit, didFailWithError error: OTError) {
+        print("Publisher failed: \(error.localizedDescription)")
+    }
+}
+
+// MARK: - OTSubscriber delegate callbacks
+extension ChatViewController: OTSubscriberDelegate {
+    func subscriberDidConnect(toStream subscriberKit: OTSubscriberKit) {
+        print("Subscriber connected")
+    }
+    
+    func subscriber(_ subscriber: OTSubscriberKit, didFailWithError error: OTError) {
+        print("Subscriber failed: \(error.localizedDescription)")
+    }
+    
+    func subscriberVideoDataReceived(_ subscriber: OTSubscriber) {
+    }
+}
+
+// MARK: - OTSubscriberKitNetworkStatsDelegate callbacks
+extension ChatViewController: OTSubscriberKitNetworkStatsDelegate {
+    
+    /**
+     * **Sender stats step 3**
+     * videoNetworkStatsUpdated method needs to be implemented
+     * to capture information about sender stats as follows
+     */
+    func subscriber(_ subscriber: OTSubscriberKit, videoNetworkStatsUpdated stats: OTSubscriberKitVideoNetworkStats) {
+        guard let senderStats = stats.senderStats else { return }
```

Comment:
Nit. If there’s another iteration in this PR, it might be worth adding a comment here noting that stats could still be nil due to network issues or latency, even if the publisher’s sender stats track is enabled.